.github/workflows/publish.yml

```
@@ -0,0 +1,36 @@
+name: Publish
+
+on:
+  workflow_dispatch:
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    name: Release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+
+      - name: Set node version to 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+
+      - name: Prepare
+        run: pnpm install --frozen-lockfile
+
+      - name: Set publishing config
+        run: pnpm config set '//registry.npmjs.org/:_authToken' "${NODE_AUTH_TOKEN}"
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

IMO we should keep the variable name consistent.

```suggestion
          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN}}
```

---

Please note that this is coming from the official GitHub documentation and I don't know if there is anything specific behind that

https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

---

> I don't know if there is anything specific behind that

Infinite copy and paste + npm uses the env variable directly, but pnpm does not.
The secret name is entirely up to the user.